src/sagemaker/estimator.py

```diff
@@ -614,16 +614,24 @@ def __init__(
         self.output_kms_key = resolve_value_from_config(
             output_kms_key, TRAINING_JOB_KMS_KEY_ID_PATH, sagemaker_session=self.sagemaker_session
         )
+        use_volume_kms_config: bool = False
         if instance_type is None or isinstance(instance_type, str):
             instance_type_for_volume_kms = instance_type
         elif isinstance(instance_type, ParameterString):
             instance_type_for_volume_kms = instance_type.default_value
         else:
-            raise ValueError(f"Bad value for instance type: '{instance_type}'")
+            # Always attach a KMS key if an instance type that isn't a string or parameter
+            # string is used.
+            use_volume_kms_config = True
+            if not isinstance(instance_type, PipelineVariable):
+                logger.warning("Unrecognized value for instance type: '%s'", instance_type)
```

Comment:
Can we add more words here in case the warning confuses customers. Meanwhile, the '%s' will invoke `__str__` method of the PipelineVariable, which can throw an error.

Can we change the warning to something like the following. Feel free to rephrase.
```
The instance_type is a PipelineVariable which can only be interpreted during pipeline execution time. Though attaching a KMS key here, please be aware that any unsupported instance_type runtime value may fail the execution.
```